### PR TITLE
Prettier merge_queue.md

### PR DESCRIPTION
**Pull request by Piinks**
Spruced up a bit, added mention of edge case for older PRs.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: docs/infra/merge_queue.md [hunk 1]
```diff
@@ -2,10 +2,15 @@
 
 If you are not hacking on flutter/flutter, you can stop reading now.
 
-tl;dr GitHub merge queue will be enabled in the flutter/flutter repo, which slightly changes the PR merge process, but you should not notice a change in your workflow.
+> [!IMPORTANT]
+> __tl;dr__ GitHub merge queue will be enabled in the flutter/flutter repo, which slightly changes the PR merge process, but you should not notice a change in your workflow.
```
Comment (ditman) [inline docs/infra/merge_queue.md:6]:
```suggestion
> **TL;DR** GitHub merge queue will be enabled in the `flutter/flutter` repo, which slightly changes the PR merge process, but you should not notice a change in your workflow.
```

File: docs/infra/merge_queue.md [hunk 2]
```diff
@@ -16,19 +21,36 @@ With the merge queue, the button will now look like this:
...
-Yes! Feel free to continue using the autosubmit label as normal. The bots will run your PR through the merge queue and land it automatically.
-What's "Remove from queue"? Can I click it?
+
+---
+
+### Can I still use autosubmit?
+
+Yes! Feel free to continue using the `autosubmit` label as normal. The bots will run your PR through the merge queue and land it automatically.
+
+If you have an older PR from before this change, `autosubmit` may refuse to merge ot. Resync your pull request to resolve this issue.
```
Comment (ditman) [inline docs/infra/merge_queue.md:31]:
```suggestion
If you have an older PR from before this change, `autosubmit` may refuse to merge it. Resync your pull request to resolve this issue.
```

File: docs/infra/merge_queue.md [hunk 3]
```diff
@@ -16,19 +21,36 @@ With the merge queue, the button will now look like this:
...
 
-WARNING: please only use this button in case of emergency. Adding and removing your PR from the queue costs in CI resources. Any CI work being done for your PR and all PRs that follow yours will be stopped and restarted.
-What if I have questions?
-If you have any questions or concerns, or If you suspect you are unable to land your PRs for reasons other than the familiar test failures and flakes, please contact any of: @yjbanov, @jtmcdole, @christopherfujino, @Piinks.
-Why do we need the merge queue?
-The main reason for this change is that flutter/engine and flutter/flutter repos will be combined into one "monorepo". In order to preserve the current lightweight dev cycle for the framework code, we need to provide pre-built engine artifacts. This way you don't need to compile any C++ code, or install extra tooling (e.g. depot_tools), when hacking on the framework alone. The merge queue is what will be building those engine binaries.
+> [!WARNING]
+> Please only use this button in case of emergency. Adding and removing your PR from the queue costs in CI resources. Any CI work being done for your PR and all PRs that follow yours will be stopped and restarted.
```
Comment (ditman) [inline docs/infra/merge_queue.md:44]: (Scary feature :P)

File: docs/infra/merge_queue.md [hunk 4]
```diff
@@ -16,19 +21,36 @@ With the merge queue, the button will now look like this:
...
+
+---
+
+### What if I have questions?
+
+If you have any questions or concerns, or If you suspect you are unable to land your PRs for reasons other than the familiar test failures and flakes, please contact any of: [@yjbanov](https://github.com/yjbanov), [@jtmcdole](https://github.com/jtmcdole), [@christopherfujino](https://github.com/christopherfujino), [@Piinks](https://github.com/piinks).
```
Comment (ditman) [inline docs/infra/merge_queue.md:50]: I don't think there's any messaging in github that would allow people to contact users directly. Should this point people to a discord chat room where the FAQ can be answered?

Also, since this document reads like a FAQ, I would move this "What if I have **other** questions?" to the bottom, so people read the whole document before resorting to direct contact.

File: docs/infra/merge_queue.md [hunk 5]
```diff
@@ -16,19 +21,36 @@ With the merge queue, the button will now look like this:
...
+
+### Why do we need the merge queue?
+
+The main reason for this change is that flutter/engine and flutter/flutter repos will be combined into one "monorepo". In order to preserve the current lightweight dev cycle for the framework code, we need to provide pre-built engine artifacts. This way you don't need to compile any C++ code, or install extra tooling (e.g. `depot_tools`), when hacking on the framework alone. The merge queue is what will be building those engine binaries.
```
Comment (ditman) [inline docs/infra/merge_queue.md:54]:
```suggestion
The main reason for this change is that `flutter/engine` and `flutter/flutter` repos will be combined into one "monorepo". In order to preserve the current lightweight dev cycle for the framework code, we need to provide pre-built engine artifacts. This way you don't need to compile any C++ code, or install extra tooling (e.g. `depot_tools`), when hacking on the framework alone. The merge queue is what will be building those engine binaries.
```

File: docs/infra/merge_queue.md [hunk 6]
```diff
@@ -2,10 +2,15 @@
 
 If you are not hacking on flutter/flutter, you can stop reading now.
 
-tl;dr GitHub merge queue will be enabled in the flutter/flutter repo, which slightly changes the PR merge process, but you should not notice a change in your workflow.
+> [!IMPORTANT]
+> __tl;dr__ GitHub merge queue will be enabled in the flutter/flutter repo, which slightly changes the PR merge process, but you should not notice a change in your workflow.
+
+On Friday November 15, 2024, GitHub merge queue feature will be enabled in the flutter/flutter repo. After this change, pull requests will first need to pass merge queue tests before landing on the target branch (e.g. `main` or `master`). At this time, no additional tests will run in the merge queue. A PR that passes presubmit checks will be allowed to land on the target branch immediately.
```
Comment (ditman) [inline docs/infra/merge_queue.md:8]:
```suggestion
On Friday November 15th, 2024, GitHub merge queue feature will be enabled in the `flutter/flutter` repo. After this change, pull requests will first need to pass merge queue tests before landing on the target branch (e.g. `main` or `master`). At this time, no additional tests will run in the merge queue. A PR that passes presubmit checks will be allowed to land on the target branch immediately.
```

File: docs/infra/merge_queue.md [hunk 7]
```diff
@@ -16,19 +21,36 @@ With the merge queue, the button will now look like this:
...
 Unlike "Squash and merge", the new button does not immediately land the PR on the target branch, but instead enqueues your PR for extra checks in the merge queue. If you do not do anything, eventually your PR will land.
-Can I still use autosubmit?
-Yes! Feel free to continue using the autosubmit label as normal. The bots will run your PR through the merge queue and land it automatically.
-What's "Remove from queue"? Can I click it?
+
+---
+
+### Can I still use autosubmit?
+
+Yes! Feel free to continue using the `autosubmit` label as normal. The bots will run your PR through the merge queue and land it automatically.
```
Comment (ditman) [inline docs/infra/merge_queue.md:29]: Should we *encourage* people to use `autosubmit` instead of manually clicking the button?

**Review by ditman — APPROVED** [review]
LGTM!

File: docs/infra/merge_queue.md (hunk 7 above)
Comment (Piinks) [reply docs/infra/merge_queue.md:29]: Folks already should be using the autosubmit label. Merging manually has been and is still discouraged.

File: docs/infra/merge_queue.md (hunk 4 above)
Comment (Piinks) [reply docs/infra/merge_queue.md:50]: We have other docs that refer to Gtihub handles for contact. I think the expectation has been that they will ping on an issue, including if that issue is a question that needs answering. :)
